### PR TITLE
log loss per image

### PR DESCRIPTION
**Pull request by noskill**
# What does this PR do?

logging loss for each image in dataset. Given that Lora datasets are usually small this feature might help to one to spot low-quality images or bad captions etc.

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_lora_sdxl.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import logging
 import math
 import os
+import os.path
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora_sdxl.py:22]: We already have `os` module imported. I think we can reuse it.

File: examples/text_to_image/train_text_to_image_lora_sdxl.py [hunk 2]
```diff
@@ -1073,10 +1078,13 @@ def compute_time_ids(original_size, crops_coords_top_left):
                     loss = loss.mean(dim=list(range(1, len(loss.shape)))) * mse_loss_weights
                     loss = loss.mean()
 
+                for fname in batch['filenames']:
+                    accelerator.log({'loss_for_' + fname: loss}, step=global_step)
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora_sdxl.py:1082]: Should we accept a CLI argument for this? That will be nice no?

**Review by sayakpaul — COMMENTED** [review]
I am generally okay with this direction. Let's maybe not introduce the unnecessary spacing changes?

File: examples/text_to_image/train_text_to_image_lora_sdxl.py (hunk 2 above)
Comment (noskill) [reply examples/text_to_image/train_text_to_image_lora_sdxl.py]: I agree, it's nice to have.

**Review by sayakpaul — APPROVED** [review]
LGTM! Let's fix the code quality issues.

**Comment by noskill** [comment]
@sayakpaul Which code quality issues? What should i fix?

**Comment by sayakpaul** [comment]
The CI says:

image

To be more specific the following step in the CI fails:

> Fast tests for PRs / check_code_quality (pull_request)

You need to run `make style && make quality` to fix these.

File: examples/text_to_image/train_text_to_image_lora_sdxl.py [hunk 3]
```diff
@@ -414,6 +414,12 @@ def parse_args(input_args=None):
         default=4,
         help=("The dimension of the LoRA update matrices."),
     )
+    parser.add_argument(
+        "--debug-loss",
+        action="store_true",
+        default=False,
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora_sdxl.py:420]: We don't need to set a default for boolean flag CLI args. If someone doesn't pass `--debug-loss` it will be defaulted to False. Also, we use "_" rather than "-" even for CLI args.

**Review by sayakpaul — COMMENTED** [review]
Thanks! Just one comment. Once addressed, will ship!

**Review by sayakpaul — APPROVED** [review]
Clean!

**Comment by noskill** [comment]
Great! Could you please merge the PR?

**Comment by sayakpaul** [comment]
Let's go!
